### PR TITLE
Refactoring and a DAO unit test

### PR DESCRIPTION
**Pull request by muralimohan123**
Refactoring to make it easy to access spring context from tests.

added a simple DAO unit test

more work to be done

### CODE REVIEW

**Comment by njacobs5074** [comment]
Murali: Did you grab the latest code before submitting this request?  GitHub is saying that there are conflicts that require manual merging - see above.

Thanks,

Nick

**Comment by muralimohan123** [comment]
Hi Nick,

I did do that. But I can also see the merge problems.

Let me try again.

-Murali

**Comment by muralimohan123** [comment]
Nick,

Sorry for the trouble, please check again.

**Comment by njacobs5074** [comment]
Looks good.
